plotly_resampler/figure_resampler/figure_resampler_interface.py

```diff
@@ -686,6 +687,36 @@ def _parse_get_trace_props(
                 if isinstance(hf_hovertext, np.ndarray):
                     hf_hovertext = hf_hovertext[not_nan_mask]
 
+            # Try to parse the hf_x data if it contains string values
+            if hf_x.dtype.type is np.str_ or hf_x.dtype == "object":
+                if len(hf_x) and isinstance(hf_x[0], str):
+                    try:
+                        # Try to parse to numeric
+                        hf_x = pd.to_numeric(hf_x, errors="raise")
+                    except ValueError:
+                        try:
+                            # Try to parse to datetime
+                            hf_x = np.asarray([pd.Timestamp(x) for x in hf_x])
+                        except ValueError:
+                            raise ValueError(
+                                "plotly-resampler requires the x-data to be numeric or datetime-like"
+                            )
+            # Check and update timezones of the hf_x data when there are multiple
+            # timezones in the data
+            if hf_x.dtype == "object":
+                if len(hf_x) and isinstance(hf_x[0], pd.Timestamp):
+                    # Assumes that all values in hf_x are pd.Timestamps
+                    if len(set(x.tz for x in hf_x)) > 1:
+                        # Remove the timezone data for plotting when multiple timezones
+                        warnings.warn(
+                            "x-data of multiple timezones / fixedoffsets is passed, "
+                            + "omitting the timezone data for plotting",
+                            UserWarning,
+                        )
+                        hf_x = np.asarray(
+                            list(map(lambda x: x.replace(tzinfo=None), hf_x))
+                        )
```

Comment:
There is also another `dtype` option missing, which is the most common but was not present in my original example due to my simplifications. It is the case when all `hf_x` is of the same TZ, or equivalently `dtype == datetime`. I haven't checked the spelling of the condition, but I am sure the type becomes a datetime64 something. So you don't treat it because it doesn't need conversion, or maybe I have just not found it?

Reply:
We missed it :see_no_evil: (see reply above)

Reply:
this is also what I tried to tackle in on the `datetime_improvements` branch, but as this PR will handle this, I will delete that branch.